src/app/features/tech-record/components/tech-record-amend-vin/tech-record-amend-vin.component.ts

```diff
@@ -0,0 +1,92 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { GlobalError } from '@core/components/global-error/global-error.interface';
+import { GlobalErrorService } from '@core/components/global-error/global-error.service';
+import { DynamicFormService } from '@forms/services/dynamic-form.service';
+import { CustomFormControl, FormNodeTypes, FormNodeWidth } from '@forms/services/dynamic-form.types';
+import { TechRecordModel, VehicleTechRecordModel } from '@models/vehicle-tech-record.model';
+import { Actions, ofType } from '@ngrx/effects';
+import { TechnicalRecordService } from '@services/technical-record/technical-record.service';
+import { updateTechRecordsSuccess } from '@store/technical-records';
+import { take } from 'rxjs';
+
+@Component({
+  selector: 'app-change-amend-vin',
+  templateUrl: './tech-record-amend-vin.component.html'
+})
+export class AmendVinComponent implements OnInit {
+  vehicle?: VehicleTechRecordModel;
+  techRecord?: TechRecordModel;
+
+  form = new FormGroup({
+    vin: new CustomFormControl(
+      {
+        name: 'input-vin',
+        label: 'Vin',
+        type: FormNodeTypes.CONTROL
+      },
+      '',
+      [Validators.minLength(3), Validators.maxLength(21), Validators.required]
+    )
+  });
+
+  constructor(
+    private actions$: Actions,
+    public dfs: DynamicFormService,
+    private globalErrorService: GlobalErrorService,
+    private route: ActivatedRoute,
+    private router: Router,
+    private technicalRecordService: TechnicalRecordService
+  ) {
+    this.technicalRecordService.selectedVehicleTechRecord$.pipe(take(1)).subscribe(vehicle => (this.vehicle = vehicle));
+
+    this.technicalRecordService.editableTechRecord$.pipe(take(1)).subscribe(techRecord => (this.techRecord = techRecord));
+  }
+
+  get width(): FormNodeWidth {
+    return FormNodeWidth.L;
+  }
+
+  ngOnInit(): void {
+    if (!this.techRecord) {
+      this.navigateBack();
+    }
+
+    this.actions$.pipe(ofType(updateTechRecordsSuccess), take(1)).subscribe(() => this.navigateBack());
+  }
+
+  get makeAndModel(): string {
+    const c = this.techRecord;
+    if (!c?.make && !c?.chassisMake) return '';
+
+    return `${c.vehicleType === 'psv' ? c.chassisMake : c.make} - ${c.vehicleType === 'psv' ? c.chassisModel : c.model}`;
+  }
+
+  get vrm(): string | undefined {
+    return this.vehicle?.vrms.find(vrm => vrm.isPrimary === true)?.vrm;
+  }
+
+  navigateBack() {
+    this.globalErrorService.clearErrors();
+    this.router.navigate(['..'], { relativeTo: this.route });
+  }
+
+  handleSubmit(): void {
+    if (!this.isFormValid) return;
+
+    //awaiting backend logic
+    console.log('valid form');
+    this.navigateBack();
+  }
+
+  get isFormValid(): boolean {
```

Comment:
Very minor but can you please put the `ngOnInit` before the getters and the `get isFormValid()` before the methods?